firestore-tos-extension/PREINSTALL.md

```diff
@@ -0,0 +1,82 @@
+Use this extension to provides an API for adding new ackowledgement terms, retrieving the latest terms, and tracking acknowledgements.
+
+Allows users to sign ackowledgement forms as part of their application. Some common ackowledgement forms include ToS and cookie ackowledgement.
+
+## Usage
+
+The extension supports multiple use cases:
+
+- Accepting terms
+- Creating terms
+- Getting terms
+- Getting accepted terms
+
+All functions `require authentication` to run, ensure you have a valud Firebase User included in your POST requests.
+
+Managed documents are stored in a specified Cloud Firestore collection, while providing two subcollections for accesing data, these are:
+
+- acknowledgements
+- agreements
+
+Each collection has a managed section for supporting multiple instances of the extension. For example, a typical routes for agreements would be:
+
+`agreement/${EXT_ID}/{agreementId}`
+
+## Definitions
+
+`tosId`: Terms of service Id. This unique Id represents a created terms firestore document.
+
+`link`: An external http link for a user to view the associated terms and conditions.
+
+`creationDate`: Automatically generated date value on creation of a new terms document.
+
+`allowList`: A list of users emails or phone numbers. If specified, this will validate to ensure only selected users can accept the specified terms.
+
+`noticeType`: A key/value list of unique properties for the terms Firestore document. An example, would be to add unique roles such as `publisher/consumer`
+
+## Auth claims (optional)
+
+By default, all accepted documents are stored under the `acknowledgements/{UID}` sub collection. Any lookups return the relavant Firestore documents.
+
+If you would like to store the accepted documents on the Firebase User as auth claims, this can be configured by enabling the `Include Authentication Claims` in the Extension configuration.
```

Comment:
What claims key is used? And what does the data structure look like?

Reply:
The claims key is currently the extensions installation name, eg: `terms-of-service-xw40`.